websocket-chat-broker/src/main.rs

```diff
@@ -0,0 +1,156 @@
+#[macro_use]
+extern crate actix;
+extern crate actix_web;
+extern crate actix_broker;
+extern crate futures;
+extern crate rand;
+
+use actix::prelude::*;
+use actix::fut;
+use actix_web::server::HttpServer;
+use actix_web::{fs, http, ws, App, Error, HttpRequest, HttpResponse};
+use actix_broker::BrokerIssue;
+use futures::Future;
+
+mod server;
+use server::*;
+
+fn chat_route(req: &HttpRequest<()>) -> Result<HttpResponse, Error> {
+    ws::start(req, WsChatSession::default())
+}
+
+#[derive(Default)]
+struct WsChatSession {
+    id: usize,
+    room: String,
+    name: Option<String>,
+}
+
+impl WsChatSession {
+    fn join_room(&mut self, room_name: &str, ctx: &mut ws::WebsocketContext<Self>) {
+        let room_name = room_name.to_owned();
+        // First send a leave message for the current room
+        let leave_msg = LeaveRoom(self.room.clone(), self.id);
+        self.issue_sync(leave_msg, ctx);
```

Comment:
Hmmm strangely enough I cannot find the `issue_sync` method, could you please point from where it is coming?

Reply:
This comes from having the `actix_broker::BrokerIssue` trait in scope. 
It implements the following functions for actors with an `AsyncContext`:
* `issue_async` - sends a message asynchronously to the message broker (uses `do_send` on the broker address from the system registry).
* `issue_sync` - sends a message synchronously to the message broker (uses `.wait(ctx)`).

After seeing your other comment about not blocking, it made me realise that maybe a third function needs to be added that returns an `ActorFuture`. This can then have computation chained on, such as the 'join room' future.

After seeing your comment about the